types.go

```diff
@@ -0,0 +1,58 @@
+package fq
+
+const (
+	scaledOne uint64 = 1 << 16
+)
+
+// mods for our algo
+// use min heap vs selectQueue
+// 1) we are dispatching requests to be served rather than packets to be transmitted
+// 2) the actual service time (i.e., duration) is not known until a request is done being served
+//
+// 1 & 2 can be handled by using duration time instead of size
+
+type Packet struct {
+	// request   http.Request
+	item      interface{}
+	virfinish uint64
+	size      uint64
+	queue     *Queue
+	//
+	key uint64
+	seq uint64
+}
+
+type Queue struct {
+	Packets       []*Packet
+	key           uint64
```

Comment:
I see no need for a Queue to have a `key`.  In almost all tests, `key` equals the queue's position in the slice of queues and so does not need to be explicitly stored.  There is only one exception, and in this case there is only one Queue so the `key` provides no distinction.